rules/set-gmsacredentialspecname-value/raw.rego

```diff
@@ -0,0 +1,45 @@
+package armo_builtins
+
+import future.keywords.if
+
+# Rule to check if the GMSA credential spec name is set in the security context
+deny[msga] {
+    # Verify the object kind
+    obj := input[_]
+    allowed_kinds := {"Pod", "Deployment", "ReplicaSet", "DaemonSet", "StatefulSet", "Job", "CronJob"}
+    allowed_kinds[obj.kind]
+
+    # Check if the GMSA credential spec name is set in the security context
+    not gmsaCredentialSpecNameSet(getSecurityContext(obj))
+
+    path := sprintf("%v.securityContext.gmsaCredentialSpecName", [obj.kind])
+    msga := {
+        "alertMessage": sprintf("%v: %v does not set 'securityContext.gmsaCredentialSpecName' with allowed value", [obj.kind, obj.metadata.name]),
+        "packagename": "armo_builtins",
+        "alertScore": 7,
+        "failedPaths": [path],
+        "fixPaths": [],
+        "alertObject": {
+            "k8sApiObjects": [obj]
+        }
+    }
+}
+
+# Function to check if the GMSA credential spec name is set in the security context
+gmsaCredentialSpecNameSet(securityContext) := true if {
```

Comment:
You should check both for `PodSecurityContext` and `SecurityContext`. What I mean is that `windowsOptions` can be found both in these places, so you should check in both of them. You can find an example here: https://github.com/kubescape/regolibrary/blob/dev/rules/set-seLinuxOptions/raw.rego.
`seLinuxOptions` can be found in both the paths. So, the path variable that we are initializing, should have the right path according to which object has been checked.

Additionally, the check here, can be simplified using the following form:
```
gmsaCredentialSpecNameSet(securityContext) := true {
    securityContext.windowsOptions.gmsaCredentialSpecName != ""
} else := false
```